CHANGELOG.md

```diff
@@ -1,11 +1,21 @@
 <!-- https://developers.home-assistant.io/docs/add-ons/presentation#keeping-a-changelog -->
 
-## 0.0.11
+## 0.1.0
```

Comment:
ok, though this will conflict with PR #78

Reply:
It's fine .. we can manually fix 1 line ;)  I'm starting to get used to it :D

I thought we were releasing 0.1.0 , that's why I change it but whatever we decide it's fine with me.

Reply:
```This is quite a huge PR```

Yes it sure is but if you click on each commit, you can view the break downs.

Things to know....
1) when I did the sorting for listen_to_mqtt... I forgot to change "send_command" to teslaCtrlSendCommand - that was done when I noticed in another PR later.
2) once I was done... had to fix shellcheck and shfmt checks.
3) to push back to origin, I had to pull fix some hard conflicts & merge

The changes are:
- functions rename
- add return code where it was needed; if I recall properly, added to sendKey, deployKey, teslaCtrlSendCommand
- use the return code properly
- add logic for what shows up or not in the device's panel
- add Tesla functions to detect if the key was sent AND *accepted*:
```
pingVehicle (key is required, this is to test if the key is valid, we expect the car to be awake since we have confimation the pub key has been delivered to the car)
acceptKeyConfirmationLoop (it loops for 5 minutes calling pingVehicle)
teslaCtrlSendKey (this is the tesla-control add-key-request)
deployKey (the main function calling teslaCtrlSendKey & acceptKeyConfirmationLoop)
```

None of the above is rocket science; the biggest changes are the commands to deploy the key and have confirmation. This should be part of your tests.